File: lectures/lect2/Prototype/src/Car.java
public class Car implements Cloneable {
    final private String mark;
    final private int power;
    final private int torque;

    public Car(String mark, int power, int torque) {
        this.mark = mark;
        this.power = power;
        this.torque = torque;
    }

    public Car(Car other) {
        this.mark = other.mark;
        this.power = other.power;
        this.torque = other.torque;
    }

    @Override
    public String toString() {
        return "Car{" +
                "mark='" + mark + '\'' +
                ", power=" + power +
                ", torque=" + torque +
                '}';
    }

    @Override
    public Car clone() {
        return new Car(this);
    }
}

File: Cpp/lab05_Prototype/task_2_1/src/car.cpp
#include "car.hpp"


Car::Car(std::string const& mark, int power, int torque)
: mark(mark), power(power), torque(torque) {}

Car::Car(Car const& other)
: mark(other.mark), power(other.power), torque(other.torque) {}

Car* Car::clone() const
{
  return new Car(*this);
}

std::ostream& operator<<(std::ostream& os, Car const& car)
{
  os << "Car{mark=" << car.mark
     << ", power=" << car.power
     << ", torque=" << car.torque
     << "}";

  return os;
}
